astropy/coordinates/baseframe.py

```diff
@@ -945,6 +945,32 @@ def is_frame_attr_default(self, attrnm):
         """
         return attrnm in self._attr_names_with_defaults
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object is the same frame as the ``other`` object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes.  Note that it does *not* matter what, if any,
+        data either object has.
+
+        Parameters
+        ----------
+        other : BaseCoordinateFrame
+            the other frame to check
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if self.__class__ == other.__class__:
+            for frame_attr_name in self.get_frame_attr_names():
```

Comment:
I think the current approach is easier for less experienced people to read (which is my guideline *if* it's not a performance issue).  It's also nominally more efficient this way because it will drop out of the loop to return False instead of going through even if one of them is False (as it would in your approach).  But that's probably a matter of microseconds, so it's a minor thing.

So I'm inclined to leave it, but I don't care too much, so if anyone else thinks this is better I could change it.